src/ResourceStateBuilder.js

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019 Stanford University see LICENSE for license
+
+import shortid from 'shortid'
+
+const rdf = require('rdf-ext')
+const _ = require('lodash')
+
+/**
+ * Builds Redux state from an RDF graph
+ */
+export default class ResourceStateBuilder {
+  /**
+   * @param {Dataset} dataset the RDF graph
+   * @param {string} resourceURI URI of the resource
+   */
+  constructor(dataset, resourceURI) {
+    this.dataset = dataset
+    this.resourceURI = resourceURI
+    this.resourceState = {}
+  }
+
+  /**
+   * @return {Object} the resource represented as Redux state
+   */
+  get state() {
+    this.resourceState = this.buildResource(rdf.namedNode(this.resourceURI))
+
+    return this.resourceState
+  }
+
+  /**
+   * @param {rdf.Term} resourceTerm NamedNode or BlankNode of resource to be built
+   * @return {Object} the resource represented as Redux state
+   */
+  buildResource(resourceTerm) {
+    const thisResourceState = {}
+    // Find the resource template id. Should be only 1.
+    const rtId = this.findResourceTemplateId(resourceTerm)
+    thisResourceState[rtId] = {}
+
+    // Find the properties
+    const propertyDataset = this.findProperties(resourceTerm)
+
+    // Add each property
+    propertyDataset.forEach((quad) => {
+      const propertyURI = quad.predicate.value
+      if (quad.object.termType === 'BlankNode') {
+        thisResourceState[rtId][propertyURI] = {}
+        thisResourceState[rtId][propertyURI][shortid.generate()] = this.buildResource(quad.object)
+      } else {
+        if (!_.has(thisResourceState[rtId], propertyURI)) {
+          thisResourceState[rtId][propertyURI] = { items: [] }
+        }
+        const item = this.buildItem(quad)
+
+        thisResourceState[rtId][propertyURI].items.push(item)
+      }
+    })
+    return thisResourceState
+  }
+
+  /**
+   * @param {rdf.Term} resourceTerm NamedNode or BlankNode of resource to find
+   * @return {string} the resource template id
+   */
+  findResourceTemplateId(resourceTerm) {
```

Comment:
if some of these methods are, effectively, private methods and not part of the API of this class, can they be so denoted?  We don't have enough context to understand how this class should be expected to be used.

Reply:
https://babeljs.io/blog/2018/12/03/7.2.0

Reply:
This doesn't seem fully baked yet. I (and others) couldn't get eslint to handle the `#` syntax.

Reply:
As an alternative, I added example usage to the class docs. This makes it clear what the API of the class is.